drivers/sensor/tdk/icm42x70/icm42x70.c

```diff
@@ -377,7 +381,8 @@ static int icm42670_accel_config(struct icm42670_data *drv_data, enum sensor_att
 	return 0;
 }
 
-static int icm42670_set_gyro_odr(struct icm42670_data *drv_data, const struct sensor_value *val)
+#if CONFIG_USE_EMD_ICM42670
```

Comment:
Suggestion, to make the code more readable add these functions to `icm42670.h` and include it from `icm42x70.h`, then move the definitions to an `icm42670.c` file and guard the inclusion of it in the cmake file. I generally find that it's easier to read the code that way.